src/coffee/directives/api/models/parent/windows-parent-model.coffee

```diff
@@ -65,16 +65,16 @@ angular.module('uiGmapgoogle-maps.directives.api.models.parent')
               @onDestroy(doDelete).then =>
                 @createChildScopes() if doCreate
 
-          onDestroy:(scope) =>
+          onDestroy: (scope) =>
             super(@scope)
             _async.promiseLock @, uiGmapPromise.promiseTypes.delete, undefined, undefined, =>
-              _async.each @plurals.values(), (child) =>
-                child.destroy()
+              _async.each @plurals.values(), (child) ->
+                child.destroy(true)#this allows scope.$destroy to be called on the child with out recursion issues
```

Comment:
This was the main bug as by not passing true the child was not getting cleaned up in scope.